gnomad/utils/constraint.py

```diff
@@ -372,6 +372,47 @@ def _get_criteria(i: hl.expr.Int32Expression) -> hl.expr.Int32Expression:
     return hl.agg.array_sum(hl.map(_get_criteria, sorted_indices))
 
 
+def explode_downsamplings(
+    ht: hl.Table,
+    downsampling_meta: Dict[str, List[str]],
+    metrics: List[str] = ["syn", "lof", "mis"],
+) -> hl.Table:
+    """
+    Explode downsampling counts for each genetic ancestry group and metric.
+
+    :param ht: Input Table.
+    :param metrics: List of metrics to explode. Default is '['syn', 'lof', 'mis']'.
+    :param downsampling_meta: Dictionary containing downsampling metadata. Keys are the genetic ancestry group names and values are the list of downsamplings for that genetic ancestry group.
+    :return: Table with downsampling counts exploded.
+    """
+    ht = ht.annotate(
+        data=[
+            hl.struct(
+                gen_anc=pop,
+                downsampling=downsampling,
+                **{
+                    f"{metric}.exp": ht[metric]["gen_anc_exp"][pop][i]
+                    for metric in metrics
+                },
+                **{
+                    f"{metric}.obs": hl.if_else(
+                        hl.is_missing(ht[metric]["gen_anc_obs"][pop][i])
+                        & hl.is_defined(ht[metric]["gen_anc_obs"][pop][i]),
+                        0,
+                        ht[metric]["gen_anc_obs"][pop][i],
+                    )
+                    for metric in metrics
+                },
+            )
+            for pop, downsamplings in downsampling_meta.items()
+            for i, downsampling in enumerate(downsamplings)
+        ]
+    ).select(*["data"])
```

Comment:
This only need to be this right?

```suggestion
    ).select("data")
```

And actually, if just selecting, that can be done instead of `annotate` above right?